### PR TITLE
Speed up slow user export functionality

### PR DESCRIPTION
**Pull request by boffbowsh**
Trello: https://trello.com/c/AbFXHJ5u/104-504-timeout-when-trying-to-download-a-csv-of-signon-users-medium

The "Export as CSV" button in signon will generally result in a 504 in production due to it taking over 15 seconds to generate the CSV (actually around 10 minutes on a VM). This is mostly avoidable by fetching data in more optimised ways rather than relying on ActiveRecord's association magic.

CSVs with no filtering are now returned in around 4 seconds on a VM, which should be quick enough.

This pull request best reviewed commit by commit.

Possible future refactoring would be to pass a scope of users into the presenter and have it generate the CSV itself.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -58,6 +58,25 @@ class User < ActiveRecord::Base
   scope :last_signed_in_after, lambda { |date| web_users.not_suspended.where('date(current_sign_in_at) >= date(?)', date) }
   scope :not_recently_unsuspended, lambda { where(['unsuspended_at IS NULL OR unsuspended_at < ?', UNSUSPENSION_GRACE_PERIOD.ago]) }
 
+  scope :with_status, lambda { |status|
+    case status
```
Comment (benilovj) [inline app/models/user.rb]: if a new status is added (eg for 2sv) and we forget to update this code (which is quite likely), this statement will not filter correctly. Is it worth adding a final condition that sends an exception to Errbit when the status hasn't been matched?

File: lib/devise/models/password_expirable.rb [hunk 2]
```diff
@@ -11,6 +11,19 @@ module PasswordExpirable
 
       included do
         before_save :update_password_changed
+
+        scope :with_need_change_password, -> do
+          if self.expire_password_after.is_a?(Fixnum) || self.expire_password_after.is_a?(Float)
```
Comment (benilovj) [inline lib/devise/models/password_expirable.rb]: The expression `self.expire_password_after.is_a?(Fixnum) || self.expire_password_after.is_a?(Float)` appears 3 times in this module, and it's not the most readable thing in the world. Is it worth extracting it into a well-named method?

File: app/models/user.rb (hunk 1 above)
Comment (boffbowsh) [inline app/models/user.rb]: I think it's probably worth just raising an exception completely. Good spot.

**Comment by benilovj** [comment]
Apart from the nitpicks, seems reasonable to me.

File: app/models/user.rb (hunk 1 above)
Comment (benilovj) [inline app/models/user.rb]: @fofr just made an excellent suggestion to write a unit test that loops through `User::USER_STATUSES` and makes sure that all statuses are covered

**Comment by boffbowsh** [comment]
Comments addressed and rebased

**Pull request merged**